app/controllers/CredentialsAuthController.scala

```diff
@@ -21,26 +21,42 @@ import play.api.libs.json.Json
 import play.api.mvc.{Action, RequestHeader}
 import play.api.Play
 import scala.concurrent.Future
+import models.daos.slick.DBTableDefinitions.{DBUser, UserTable}
 
 /**
-  * The credentials auth controller that is responsible for user log in.
-  *
-  * @param env The Silhouette environment.
-  */
+ * The credentials auth controller that is responsible for user log in.
+ *
+ * @param env The Silhouette environment.
+ */
 class CredentialsAuthController @Inject() (
                                             implicit val env: Environment[User, SessionAuthenticator],
                                             val userService: UserService)
   extends Silhouette[User, SessionAuthenticator] with ProvidesHeader  {
 
   /**
-    * Authenticates a user against the credentials provider.
-    */
+   * Authenticates a user against the credentials provider.
+   */
   def authenticate(url: String) = Action.async { implicit request =>
+    // Logs general signin attempt
+    val ipAddress: String = request.remoteAddress
+    val timestamp: Timestamp = new Timestamp(Instant.now.toEpochMilli)
+    val anonymousUser: DBUser = UserTable.find("anonymous").get
+    WebpageActivityTable.save(WebpageActivity(0, anonymousUser.userId.toString, ipAddress, "SignInAttempt", timestamp))
```

Comment:
Everywhere where you do `anonymousUser.userId.toString`, the `.toString` is redundant, because `anonymousUser.userId` is already a string.